### PR TITLE
Features/related content

### PR DESCRIPTION
**Pull request by collingo**
@davide @ch2ch3 @phil-linnell could you take a look?

You can see related content on any case study or blog post (and eventually the homepage once integrated). It loads asynchronously so may not be there immediately.

You'll need to delete any existing local `features/related-content` branches as I have rebased a couple of times :)

### CODE REVIEW

**Comment by phil-linnell** [comment]
Fixed blog category colours and fonts. Everything looks good to me.
I will tackle the rollover states once it's merged to master as it goes with my current rollover-states branch.

:+1:

**Comment by ch2ch3** [comment]
LGTM :+1: :tada: :smile: 

May be premature on my part, but I feel that the async data fetching and applying functions could be a bit more generic. We don't seem to be doing anything complicated with related content in particular that necessitates checking if `dependency === related_content`, or having a function `applyRelatedContent` vs `applyAsyncData`.

e.g.
```
if(dependency === 'related_content') {
  DataLoader(get(_state[item.type], 'related_content', []).map(url => {
    return {
      url: url,
      type: 'relatedContent'
    };
  }), applyRelatedContent);
}
```

could be

```
DataLoader(get(_state[item.type], dependency, []).map(url => {
  return {
    url: url,
    type: camelCase(dependency)
  };
}), applyAsyncData);
```

I guess there's some debate about whether the `dependency` will always be nested within `item` that we can use `get(_state[item.type], dependency, [])` but if that's the case then the conditional should just be for getting the appropriate array of items for DataLoader rather than encompassing the whole DataLoader call.

File: src/app/flux/store.js [hunk 1]
```diff
@@ -105,16 +125,31 @@ const Store = Object.assign(
...
+      DataLoader(itemsToLoad, applyData).then(() => {
+        Store.emit('change', _state);
+        Store.initiateAsyncLoadsFor(itemsToLoad);
+      });
+    },
+    initiateAsyncLoadsFor(itemsToLoad) {
+      switch(true) {
+        case some(itemsToLoad, item => item.type === 'posts'):
+          Store.getSocialSharesForPosts();
+        break;
```
Comment (ch2ch3) [inline src/app/flux/store.js]: `some(itemsToLoad, type, 'posts')`?

Comment (ch2ch3) [inline src/app/flux/store.js]: Also indentation for L137 should be same as L136 :see_no_evil:

**Comment by collingo** [comment]
@ch2ch3 thanks for the feedback, have implemented everything you suggested.

fyi the only issue with the generic async dependency loader was the application function. I got round that using a function getter that takes the dependency and returns the correct function.

Let me know what you think...

**Comment by phil-linnell** [comment]
Some additional css + rebase, all good :+1:

**Pull request merged**